model/registry/cicd.yaml

```diff
@@ -0,0 +1,92 @@
+      version of software. This aligns with the
+      [Britannica](https://www.britannica.com/dictionary/pipeline) definition
+      of a pipeline where a **pipeline** is the system for developing and
+      producing something. In the context of CI/CD, a pipeline produces or
+      delivers software.
+    attributes:
+      - id: name
+        type: string
+        stability: experimental
+        brief: >
+          The human readable name of the pipeline within a CI/CD system.
+        examples: ['Build and Test', 'Lint', 'Deploy Go Project', 'deploy_to_environment']
+      - id: run.id
+        type: string
+        stability: experimental
+        brief: >
+          The unique identifier of a pipeline run within a CI/CD system.
+        examples: ['120912']
+      - id: task.name
+        type: string
+        stability: experimental
+        brief: >
+          The human readable name of a task within a pipeline. Task here most
+          closely aligns with a [computing process](https://en.wikipedia.org/wiki/Pipeline_(computing))
+          in a pipeline. Other terms for tasks include commands, steps, and
+          procedures.
+        examples: ['Run GoLang Linter', 'Go Build', 'go-test', 'deploy_binary']
+      - id: task.run.id
+        type: string
+        stability: experimental
+        brief: >
+          The unique identifier of a task run within a pipeline.
+        examples: ['12097']
+      - id: task.type
+        type:
+          allow_custom_values: true
+          members:
+            - id: build
+              value: build
+              brief: build
+              stability: experimental
+            - id: test
+              value: test
+              brief: test
+              stability: experimental
+            - id: deploy
+              value: deploy
+              brief: deploy
+              stability: experimental
+        stability: experimental
+        brief: >
+          The type of the task within a pipeline.
+        examples: ['build', 'test', 'deploy']
+      # TODO: May change name to type and make it a well known enum.
+      # Need definition of a runner linked.
+      - id: task.runner.name
+        type: string
+        stability: experimental
+        brief: >
+          The name of the runner executing the step within a pipeline.
+        examples: ['ubuntu-runner-latest']
+      - id: task.runner.id
+        type: string
+        stability: experimental
+        brief: >
+          The id of the runner executing the step within a pipeline.
+        examples: ['1289']
+      # TODO: Figure out how to inherit (embed) system attributes. - This maybe
+      # be WIP already taking place and simply may need to be removed for the
+      # time being.
+      - id: step.runner.system
+        type: string
+        stability: experimental
+        brief: >
+          OS system attributes of the runner acorrding to the [OS system attributes](./os.md).
```

Comment:
Are here defined the values under attributes i.e. OS name or attribute itself as in example? `os.name`? If second, what is the purpose for this field?

Reply:
This is one of the ones I haven't adjusted yet. The intent it is to inherit the OS System semantic conventions for the runner. At the original time of writing that, there wasn't a way to embed those attributes. I still need to adjust (or remove) this for now until such time I can embed such attributes.

Reply:
I was originally thinking of this as an example https://opentelemetry.io/docs/specs/semconv/system/system-metrics/#systemos---os-specific-system-metrics however this appear to be all manually written text within the metrics markdown and isn't something provided through the generator. My current thinking is that I should redefine the attributes from system, similar to how each of the vendor extensions and keep it super simple like system.os.name, system.os.type, or w/e else. 

Or maybe I should remove this for now & open up a dedicated issue for this. Thoughts?

Reply:
do I understand that here we basically need to embed other attributes from os namespace?
there is an [open PR](https://github.com/open-telemetry/weaver/pull/212) to do this, maybe you can create a dedicated issue out of it and postpone until it will be used into semconv?

Reply:
Yes, that's correct. I like this approach to remove for this PR & then open an issue for this. Will take care of that. Thanks!

Reply:
Created #1184